x-pack/metricbeat/module/gcp/vertexai_logs/vertexai_logs.go

```diff
@@ -0,0 +1,261 @@
+	meta, err := dataset.Metadata(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting dataset metadata: %w", err)
+	}
+
+	events, err := m.queryVertexAILogs(ctx, client, meta.Location)
+	if err != nil {
+		return fmt.Errorf("queryVertexAILogs failed: %w", err)
+	}
+
+	m.logger.Debugf("Total %d events created for vertexai_logs", len(events))
+	for _, event := range events {
+		reporter.Event(event)
+	}
+
+	// Update watermark with latest logging_time from events
+	m.updateLastLoggingTime(events)
+
+	return nil
+}
+
+func (m *MetricSet) queryVertexAILogs(ctx context.Context, client *bigquery.Client, location string) ([]mb.Event, error) {
+	query := m.generateQuery()
+	m.logger.Debug("bigquery query = ", query)
+
+	q := client.Query(query)
+	q.Location = location
+
+	job, err := q.Run(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("bigquery Run failed: %w", err)
+	}
+
+	status, err := job.Wait(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("bigquery Wait failed: %w", err)
+	}
+
+	if err := status.Err(); err != nil {
+		return nil, fmt.Errorf("bigquery status error: %w", err)
+	}
+
+	it, err := job.Read(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("reading from bigquery job failed: %w", err)
+	}
+
+	var rows []VertexAILogRow
+	for {
+		var row VertexAILogRow
+		err := it.Next(&row)
+		if errors.Is(err, iterator.Done) {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("bigquery RowIterator Next failed: %w", err)
+		}
+		rows = append(rows, row)
+	}
+
+	events := EventsMapping(rows, m.config.ProjectID, m.logger)
+	return events, nil
+}
+
+func (m *MetricSet) generateQuery() string {
+	escapedTableID := fmt.Sprintf("`%s`", m.config.TableID)
+
+	var whereClause string
+	if m.lastLoggingTime != nil {
+		// Incremental query: get records after last processed time
+		whereClause = fmt.Sprintf("logging_time > TIMESTAMP('%s')",
+			m.lastLoggingTime.Format("2006-01-02 15:04:05.000000"))
+		m.logger.Debugf("Using incremental query from logging_time: %s", m.lastLoggingTime.Format(time.RFC3339))
+	} else {
+		// First run: use timestamp filter with lookback
+		whereClause = fmt.Sprintf("logging_time >= TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL %d HOUR)",
+			m.config.TimeLookbackHours)
+		m.logger.Debugf("Using initial query with timestamp filter: %d hours", m.config.TimeLookbackHours)
+	}
+
+	query := fmt.Sprintf(`
+SELECT
+	IFNULL(endpoint, '') AS endpoint,
+	IFNULL(deployed_model_id, '') AS deployed_model_id,
+	logging_time,
+	IFNULL(CAST(request_id AS STRING), '') AS request_id,
+	IFNULL(request_payload, []) AS request_payload,
+	IFNULL(response_payload, []) AS response_payload,
+	IFNULL(model, '') AS model,
+	IFNULL(model_version, '') AS model_version,
+	IFNULL(api_method, '') AS api_method,
+	IFNULL(TO_JSON_STRING(full_request), '{}') AS full_request,
+	IFNULL(TO_JSON_STRING(full_response), '{}') AS full_response,
+	IFNULL(TO_JSON_STRING(metadata), '{}') AS metadata,
+	IFNULL(TO_JSON_STRING(otel_log), '{}') AS otel_log
```

Comment:
what's the plan with this field? it sounds like if this is enabled it contains the entire log message in otel format? shouldn't we do one or the other?

Reply:
if we're not using it and we don't know what it contains we should remove it for now. we can always add it later.